### ISSUE TITLE
Expand Require Namespace Filtering to projects and multiple projects/namespaces

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#7618]
Original Issue - https://github.com/rancher/dashboard/issues/6483

Once the BE team update API filtering to cover projects and multiple namespaces this feature should be updated to use them. I don't think that work has a separate ticket, but is done via the pagination work

**Comment by richard-cox** [comment]
This will be achievable once https://github.com/rancher/rancher/issues/40141 has been completed

**Comment by gaktive** [comment]
Related backend is now To Test so UI unblocked.

**Comment by gaktive** [comment]
https://github.com/rancher/rancher/issues/40140 is undergoing some revision; once that's sorted, this should unblock UI.

**Comment by floatingman** [comment]
Tested the issue by performing the following steps

1. Enabling the updated performance setting in Global Settings (Advanced Websocket Worker, Required Namespace/ Project Filtering)
2. Verified the following
    1. All workloads list fetch the correct data (via http request)
    2. New endpoint is hit for primary resource type, only resources from selected namespaces/projects are shown
    3. New endpoint is hit for all secondary resource types
    4. All workloads list update with the correct data (via socket)
    5. Changes to resources in the list are automatically shown (when changed via another context/browser/etc)
    6. Changes to resource not in the list do not show (creating a new resource not in a selected namespace/project doesn't appear in the list)
9. Creating resources works fine
10. Navigate to a create page of a resource (deployment, pod). Resource can be created, user is returned to the resource list and, if the resource was created in a namespace in the filter, the new resource should be shown
11. Edit resources work fine
12. Navigate to an existing resource page (deployment, pod). Resource can be edited, user is returned to the resource list and the edited resource should be shown
13. Both Upstream and Downstream clusters work correctly

**Comment by richard-cox** [comment]
Perfect, thanks @floatingman!
